vars/publishSmokeTestResults.groovy

```diff
@@ -0,0 +1,256 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/** Library to fetch the failing Integration test details at the end of Integration Test Jenkins build and index the results to OpenSearch Metrics cluster.
+ *
+ * @param Map args = [:] args A map of the following parameters.
+ * @param args.distributionBuildUrl <required> - The jenkins distribution build number.
+ * @param args.testReportManifestYml <required> - The generated test report YAML file using test report workflow.
+ * @param args.jobName <required> - The smoke test job name, used in `testReportManifestYmlUrl`.
```

Comment:
Why do we need jobName as required? Seeing that is being only used to get `testReportManifestYmlUrl`
Would it make sense to directly set arg `testReportManifestYmlUrl` instead? 
Also what is the difference between `testReportManifestYml` and `testReportManifestYmlUrl`

Reply:
`testReportManifestYml` is the test report file which will be read through to retrieve test results.
`testReportManifestYmlUrl` is the URL which will be indexed into metrics cluster as a link to the report.